core/parse.y

```diff
@@ -255,24 +255,28 @@ expr:
 	}
 |	'(' expr for_clauses if_clauses_opt ')'
 	{
+		exprStart, _ := $2.Span()
 		$$ = &ListForExpr{
 			Brack: "()",
 			Start: $1,
 			X: $2,
 			For: $3,
 			If: $4,
 			End: End{Pos: $5},
+			ForceMultiLine: $1.Line != exprStart.Line,
```

Comment:
Automatically collapsing or expanding depending on line-length would indeed be a natural feature request. However, this seems too complicated for a drive-by contributor (i.e. me) to implement as heuristics in the presence of many possible line-breaks are likely to get complicated. In the test I have used long lines to make sure that it does not break if such a feature gets introduced.

Currently, as far as I can tell, () and {} comprehensions never get newlines after (/{, before "for" and before )/}. With this change, the user can request those newlines by adding a newline after the opening bracket.

You can have buildifier do the change from the example by deleting the newline after [.

Note that this change does not affect the behavior of []-comprehensions (including the behavior surrounding the example). It merely extends the current behavior for [... for ....] to (... for ...) and {...: ... for ...}.